packages/core/lib/services/credential/config.js

```diff
@@ -13,11 +13,22 @@ export const validation = {
     const {vcJson, keyDoc} = params;
     assert(typeof vcJson === 'object', 'invalid vcJson');
     assert(typeof keyDoc === 'object', 'invalid keyDoc');
-    assert(keyDoc.type === 'Ed25519VerificationKey2018', 'invalid type');
 
     assert(
       typeof keyDoc.publicKeyBase58 === 'string',
       'publicKeyBase58 is not present',
     );
   },
+  createPresentation: params => {
```

Comment:
The following validations should ideally be part of SDK (https://github.com/docknetwork/sdk) and then SDK should be used.